helper/resource/testing_new_import_state.go

```diff
@@ -279,7 +317,7 @@ func testStepNewImportState(ctx context.Context, t testing.T, helper *plugintest
 			identifierAttribute = "id"
 		}
 
-		for _, r := range newResources {
+		for _, r := range newResources { // This is a problem when newResources is empty!
```

Comment:
The `if !found` logic prevents false-negatives when `newResources` is empty.

I was puzzled that `ImportStateVerify` was 🍏 for some plannable import tests despite there being no apply, thus no new state.

This explains why they were trivially satisfying `ImportStateVerify`.

I'll add a guard to error early: if plannable, then ImportStateVerify is not supported.